### PR TITLE
Dell 10g adapter

### PR DESCRIPTION
**Pull request by lindycoder**
Also added a debug flag to the subshell tool so that when an error happens in a test we can see it in the log

### CODE REVIEW

File: netman/adapters/switches/dell10g.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        result = self.shell.do('show vlan')
+        return parse_vlan_list(result)
+
+    def get_interfaces(self):
+        result = self.shell.do('show interfaces status')
+        return [self.read_interface(name) for name in parse_interface_names(result)]
+
+    def add_vlan(self, number, name=None):
+        with self.config():
+            result = self.shell.do('vlan %s' % number)
```
Comment (idjaw) [inline netman/adapters/switches/dell10g.py]: should change to '{}'.format()

File: netman/adapters/switches/dell10g.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            if len(result) > 0:
+                raise BadVlanNumber()
+            else:
+                if name:
+                    result = self.shell.do('name %s' % name)
```
Comment (idjaw) [inline netman/adapters/switches/dell10g.py]: should change to '{}'.format()

File: tests/adapters/switches/dell10g_test.py [hunk 3]
```diff
@@ -0,0 +1,957 @@
...
+
+        with self.configuring_and_committing():
+            self.mocked_ssh_client.should_receive("do").with_args("interface tengigabitethernet 1/0/4").once().ordered().and_return([])
+            self.mocked_ssh_client.should_receive("do").with_args("shutdown").once().ordered().and_return([])
+            self.mocked_ssh_client.should_receive("do").with_args("exit").once().ordered().and_return([])
+
+        self.switch.shutdown_interface("tengigabitethernet 1/0/4")
+
+    def test_shutdown_interface_invalid_interface_raises(self):
+        #TODO: erreur ?????????????????????????????
```
Comment (stephanerobert) [inline tests/adapters/switches/dell10g_test.py]: Is that comment really wanted ?

File: tests/adapters/switches/dell10g_test.py [hunk 4]
```diff
@@ -0,0 +1,957 @@
...
+
+        with self.configuring_and_committing():
+            self.mocked_ssh_client.should_receive("do").with_args("interface tengigabitethernet 1/0/4").and_return([]).once().ordered()
+            self.mocked_ssh_client.should_receive("do").with_args("no shutdown").and_return([]).once().ordered()
+            self.mocked_ssh_client.should_receive("do").with_args("exit").once().and_return([])
+
+        self.switch.openup_interface("tengigabitethernet 1/0/4")
+
+    def test_openup_interface_invalid_interface_raises(self):
+        #TODO: erreur ?????????????????????????????
```
Comment (stephanerobert) [inline tests/adapters/switches/dell10g_test.py]: #TODO: Let's fix it

**Comment by lindycoder** [comment]
Comments have been addressed

**Comment by stephanerobert** [comment]
:+1:

File: tests/adapters/switches/dell10g_test.py (hunk 4 above)
Comment (idjaw) [inline tests/adapters/switches/dell10g_test.py]: Are you aware of this?

**Comment by idjaw** [comment]
:+1:

**Pull request merged**